### ISSUE TITLE
Button high contrast mode issues

### ISSUE THREAD

**Issue by khmakoto** [microsoft/fluentui#21403]
The a11y review of the v9 `Button` component (#20917) identified a bunch of high contrast mode issues that need to be fixed:

- Border change after mouse click stays active even after mouse up.
- Transparent variant does not elicit a change after `:hover` nor `:active`.
- Icons in non-primary variants are not visible.
